src/gluonts/core/component.py

```diff
@@ -339,9 +340,12 @@ def init_wrapper(*args, **kwargs):
                         if not skip_encoding(arg)
                     }
                 )
-                self.__class__.__getnewargs_ex__ = validated_getnewargs_ex
                 self.__class__.__repr__ = validated_repr
```

Comment:
Notice that we do not have to make changes to setting `__repr__` because this method is explicitly defined on every Python class by default.